packages/autorest.python/autorest/codegen/models/model_type.py

```diff
@@ -219,9 +219,22 @@ def discriminator(self) -> Optional[Property]:
         except StopIteration:
             return None
 
+    @property
+    def _has_own_discriminator(self) -> bool:
```

Comment:
we can just add a property in the emitter for these models, like `isPolymorphicSubtype`

Reply:
Hi @iscai-msft , I added a PR for adding this in emitter: https://github.com/Azure/cadl-azure/pull/1830, would you have a look?